python/ray/_private/utils.py

```diff
@@ -1492,15 +1494,47 @@ def internal_kv_put_with_retry(gcs_client, key, value, namespace, num_retries=20
     raise error
 
 
+@dataclass
+class VersionInfo:
+    ray_version: str
+    python_version: str
+    python_magic_number: Optional[str]
+
+    def __str__(self):
+        if self.python_magic_number is not None:
+            magic_str = f" (magic number {self.python_magic_number})"
+        else:
+            magic_str = ""
+        return (
+            f"    Ray: {self.ray_version}\n"
+            + f"    Python: {self.python_version}{magic_str}\n"
+        )
+
+    def __eq__(self, other):
```

Comment:
I think we shouldn't provide `__eq__` function here. Different users of `VersionInfo` may wants to compare differently based on their use cases. We should just provide the data and let users decide what they want to compare.